genesis/utils/path_planing.py

```diff
@@ -0,0 +1,995 @@
+        for i_b in envs_idx:
+            if self._rrt_is_active[i_b]:
+                is_collision_detected = False
+                if not ignore_collision:
+                    is_collision_detected = self._func_check_collision(
+                        ignore_geom_pairs,
+                        i_b,
+                        is_plan_with_obj,
+                        obj_geom_start,
+                        obj_geom_end,
+                    )
+                if is_collision_detected:
+                    self._rrt_tree_size[i_b] -= 1
+                    self._rrt_node_info[self._rrt_tree_size[i_b], i_b].configuration = 0.0
+                    self._rrt_node_info[self._rrt_tree_size[i_b], i_b].parent_idx = -1
+                else:
+                    # check the obtained steer result is within goal configuration only if no collision
+                    is_goal = True
+                    for i_q in range(self._entity.n_qs):
+                        if (
+                            ti.abs(
+                                self._solver.qpos[i_q + self._entity._q_start, i_b]
+                                - self._rrt_goal_configuration[i_q, i_b]
+                            )
+                            > self._rrt_pos_tol
+                        ):
+                            is_goal = False
+                            break
+                    if is_goal:
+                        self._rrt_goal_reached_node_idx[i_b] = self._rrt_tree_size[i_b] - 1
+                        self._rrt_is_active[i_b] = 0
+
+    def plan(
+        self,
+        qpos_goal,
+        qpos_start=None,
+        resolution=0.05,
+        timeout=None,
+        max_nodes=2000,
+        smooth_path=True,
+        num_waypoints=100,
+        ignore_collision=False,
+        ee_link_idx=None,
+        obj_entity=None,
+        envs_idx=None,
+    ):
+        if self._solver.n_envs > 0:
+            envs_idx = self._solver._sanitize_envs_idx(envs_idx)
+        else:
+            envs_idx = torch.zeros(1, dtype=gs.tc_int, device=gs.device)
+
+        qpos_cur = self._entity.get_qpos(envs_idx=envs_idx)
+        qpos_cur, qpos_goal, qpos_start, envs_idx = self.format_input(qpos_cur, qpos_goal, qpos_start, envs_idx)
+        unique_pairs = self.get_exclude_geom_pairs(qpos_goal, qpos_start, envs_idx)
+
+        is_plan_with_obj = False
+        obj_geom_start, obj_geom_end = -1, -1
+        if ee_link_idx is not None and obj_entity is not None:
+            is_plan_with_obj = True
+            obj_geom_start = obj_entity.geom_start
+            obj_geom_end = obj_entity.geom_end
+            obj_link_idx = obj_entity._links[0].idx
+            T_robot_obj = self.get_link_transformation(ee_link_idx, obj_link_idx)[envs_idx]
+
+        self._init_rrt_fields(max_nodes=max_nodes, max_step_size=resolution)
+        self._reset_rrt_fields()
+        self._kernel_rrt_init(qpos_start.contiguous(), qpos_goal.contiguous(), envs_idx)
```

Comment:
Just make sure that `format_input` returns contiguous tensors in the first place.